streaming/rrdpush.c

```diff
@@ -1396,31 +1427,36 @@ int rrdpush_receiver_thread_spawn(RRDHOST *host, struct web_client *w, char *url
             timezone = value;
         else if(!strcmp(name, "tags"))
             tags = value;
+        else if(!strcmp(name, "ver"))
+            stream_version = MIN((uint32_t) strtoul(value, NULL, 0), STREAMING_PROTOCOL_CURRENT_VERSION);
         else {
-            if(!strcmp(name, "NETDATA_PROTOCOL_VERSION"))
-                stream_flags = LABEL_FLAG_UPDATE_STREAM;
-            else {
-                // An old Netdata slave does not have a compatible streaming protocol, map to something sane.
-                if (!strcmp(name, "NETDATA_SYSTEM_OS_NAME"))
-                    name = "NETDATA_HOST_OS_NAME";
-                else if (!strcmp(name, "NETDATA_SYSTEM_OS_ID"))
-                    name = "NETDATA_HOST_OS_ID";
-                else if (!strcmp(name, "NETDATA_SYSTEM_OS_ID_LIKE"))
-                    name = "NETDATA_HOST_OS_ID_LIKE";
-                else if (!strcmp(name, "NETDATA_SYSTEM_OS_VERSION"))
-                    name = "NETDATA_HOST_OS_VERSION";
-                else if (!strcmp(name, "NETDATA_SYSTEM_OS_VERSION_ID"))
-                    name = "NETDATA_HOST_OS_VERSION_ID";
-                else if (!strcmp(name, "NETDATA_SYSTEM_OS_DETECTION"))
-                    name = "NETDATA_HOST_OS_DETECTION";
-                if (unlikely(rrdhost_set_system_info_variable(system_info, name, value))) {
-                    info("STREAM [receive from [%s]:%s]: request has parameter '%s' = '%s', which is not used.",
+            // An old Netdata slave does not have a compatible streaming protocol, map to something sane.
+            if (!strcmp(name, "NETDATA_SYSTEM_OS_NAME"))
+                name = "NETDATA_HOST_OS_NAME";
+            else if (!strcmp(name, "NETDATA_SYSTEM_OS_ID"))
+                name = "NETDATA_HOST_OS_ID";
+            else if (!strcmp(name, "NETDATA_SYSTEM_OS_ID_LIKE"))
+                name = "NETDATA_HOST_OS_ID_LIKE";
+            else if (!strcmp(name, "NETDATA_SYSTEM_OS_VERSION"))
+                name = "NETDATA_HOST_OS_VERSION";
+            else if (!strcmp(name, "NETDATA_SYSTEM_OS_VERSION_ID"))
+                name = "NETDATA_HOST_OS_VERSION_ID";
+            else if (!strcmp(name, "NETDATA_SYSTEM_OS_DETECTION"))
+                name = "NETDATA_HOST_OS_DETECTION";
+            else if(!strcmp(name, "NETDATA_PROTOCOL_VERSION") && stream_version == UINT_MAX) {
+                stream_version = 1;
+            }
+
+            if (unlikely(rrdhost_set_system_info_variable(system_info, name, value))) {
+                info("STREAM [receive from [%s]:%s]: request has parameter '%s' = '%s', which is not used.",
                      w->client_ip, w->client_port, key, value);
```

Comment:
@thiagoftsm  I still see that you put in the logfile `w->client_ip, w->client_port, key, value`. Did you mean to write the actual parameter (`name`) that is not used?

Reply:
Hi @stelfrag , I did not created this message, this is in Netdata since a long time ago, so I decided to keep it while I was organizing the code.
But, I liked your idea, I have only one doubt. Should we not create a separate PR for this?

Reply:
Ok then I don't have a strong case about fixing it now. It will report keys when a new version of the streaming protocol is out and the slave is updated before the master :+1:

Reply:
@stelfrag you mean it will log api key?

Reply:
> @stelfrag you mean it will log api key?

Yes with the api key -- I edited the comment -- thanks @ilyam8 !!

Reply:
Keep in mind that netdata error log is word readable, if api key is private it shouldnt be logged

Reply:
A very good point @ilyam8 , you convinced me to use name now.

Reply:
I changed the message using `name` instead the `api key`.
Thank you guys!

Reply:
If there are any connection errors on the stream then the api key is logged in the error message (in rrdpush.c e.g. L1462), if this is a problem then please raise an issue to fix it. I suspect the correct approach is that the error log should not be world readable.

Reply:
@amoss , this will be subject for another PR. The sre team was working days ago with installation permissions, probably we need to talk with them about it.

Reply:
That sounds good - let's have a chat within the team about the pros/cons of making the error log not world writable first. If the team thinks that it is a good idea then we can raise an issue for the SRE team to look at the file permissions.

Reply:
Dunno, anyway private info in error.log in very questionable idea. Changing perms doesnt help. We pretty often ask user to share logs and they do it private. It means that we shouldnt log any private things there. At least for now when we have one error.log for everything.